Celeste.Mod.mm/Mod/UI/OuiModTogglerProgress.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Celeste.Mod.UI {
+    public class OuiModTogglerProgress : OuiLoggedProgress {
+        public override IEnumerator Enter(Oui from) {
+            Everest.Loader.OnCrawlMod += logCrawlMod;
+            Init<OuiMainMenu>(Dialog.Clean("MODOPTIONS_MODTOGGLE_PROGRESS_TITLE"), new Task(toggleMods),
+                Everest.Loader.NewlyUnblacklistedMods.Count());
+            
+            return base.Enter(from);
+        }
+
+        public override IEnumerator Leave(Oui next) {
+            Everest.Loader.OnCrawlMod -= logCrawlMod;
+            Everest.Loader.NewlyUnblacklistedMods = null;
+            MainThreadHelper.Do(() => ((patch_OuiMainMenu) Overworld.GetUI<OuiMainMenu>())?.RebuildMainAndTitle());
+            Audio.Play(SFX.ui_main_button_back);
+            
+            return base.Leave(next);
+        }
+
+        private void toggleMods() {
+            // give it a second to transition
+            Thread.Sleep(1000);
+            int oldDelayedMods = Everest.Loader.Delayed.Count;
+            Everest.Loader.EnforceOptionalDependencies = true;
+            foreach (string mod in Everest.Loader.NewlyUnblacklistedMods) {
+                try {
+                    // remove the mod from the loaded blacklist & attempt to load mod
+                    LogLine(string.Format(Dialog.Get("DEPENDENCYDOWNLOADER_MOD_UNBLACKLIST"), mod));
```

Comment:
is reusing loc strings like this alright; should I duplicate it, rename the ID to something more generic?

Reply:
That should be fine, I don't expect that there'd be much room for confusion in other translations